predicate/Results.md

```diff
@@ -112,3 +112,15 @@ TxPredicateResults
 // bytes
 0x01, 0x02, 0x03
 ```
+
+### Maximum Size
+
+Results has a maximum size of 1MB enforced by the codec. The maximum actual size depends on how much data the available Precompile predicates may put into the results / the gas cost they charge and the block gas limit.
+
+The Results maximum size should comfortably exceed the maximum value that could happen in practice, so that a correct block builder will not attempt to build a block and fail to marshal the predicate results using the codec.
+
+We make this easy to reason about by assigning a minimum gas cost to the `PredicateGas` function of precompiles. In the case of Warp, the minimum gas cost is set to 200k gas, which can lead to at most 32 additional bytes being included in Results.
+
+The additional bytes come from the transaction hash (32 bytes), length of tx predicate results (4 bytes), the precompile address (20 bytes), length of the bytes result (4 bytes), and the additional byte in the results bitset (1 byte). This results in 200k gas contributing a maximum of 61 additional bytes to Result.
```

Comment:
I thought we are changing to indices instead of hashes?

Reply:
It is currently using hashes, so this documentation is correct as is. I don't have a strong opinion on switching to index offsets.